sdk/program/src/alt_bn128_compression.rs

```diff
@@ -0,0 +1,497 @@
+pub mod prelude {
+    pub use crate::alt_bn128_compression::{
+        alt_bn128_compression_size::*, consts::*, target_arch::*, AltBn128CompressionError,
+    };
+}
+
+use thiserror::Error;
+
+mod consts {
+    pub const ALT_BN128_COMPRESSION_G1_COMPRESS: u64 = 0;
+    pub const ALT_BN128_COMPRESSION_G1_DECOMPRESS: u64 = 1;
+    pub const ALT_BN128_COMPRESSION_G2_COMPRESS: u64 = 2;
+    pub const ALT_BN128_COMPRESSION_G2_DECOMPRESS: u64 = 3;
+}
+
+mod alt_bn128_compression_size {
+    pub const G1: usize = 64;
+    pub const G2: usize = 128;
+    pub const G1_COMPRESSED: usize = 32;
+    pub const G2_COMPRESSED: usize = 64;
+}
+
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
+pub enum AltBn128CompressionError {
+    #[error("Unexpected error")]
+    UnexpectedError,
+    #[error("Failed to decompress g1")]
+    DecompressingG1Failed,
+    #[error("Failed to decompress g2")]
+    DecompressingG2Failed,
+    #[error("Failed to compress affine g1")]
+    CompressingG1Failed,
+    #[error("Failed to compress affine g2")]
+    CompressingG2Failed,
+    #[error("Invalid input size")]
+    InvalidInputSize,
+}
+
+impl From<u64> for AltBn128CompressionError {
+    fn from(v: u64) -> AltBn128CompressionError {
+        match v {
+            1 => AltBn128CompressionError::DecompressingG1Failed,
+            2 => AltBn128CompressionError::DecompressingG2Failed,
+            3 => AltBn128CompressionError::CompressingG1Failed,
+            4 => AltBn128CompressionError::CompressingG2Failed,
+            5 => AltBn128CompressionError::InvalidInputSize,
+            _ => AltBn128CompressionError::UnexpectedError,
+        }
+    }
+}
+
+impl From<AltBn128CompressionError> for u64 {
+    fn from(v: AltBn128CompressionError) -> u64 {
+        match v {
+            AltBn128CompressionError::DecompressingG1Failed => 1,
+            AltBn128CompressionError::DecompressingG2Failed => 2,
+            AltBn128CompressionError::CompressingG1Failed => 3,
+            AltBn128CompressionError::CompressingG2Failed => 4,
+            AltBn128CompressionError::InvalidInputSize => 5,
+            AltBn128CompressionError::UnexpectedError => 0,
+        }
+    }
+}
+
+#[cfg(not(target_os = "solana"))]
+mod target_arch {
+
+    use {
+        super::*,
+        crate::alt_bn128_compression::alt_bn128_compression_size,
+        ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Compress, Validate},
+    };
+
+    type G1 = ark_bn254::g1::G1Affine;
+    type G2 = ark_bn254::g2::G2Affine;
+
+    pub fn alt_bn128_compression_g1_decompress(
```

Comment:
Nit: Let's keep this simple and use `alt_bn128_g1_decompress`. I think the compression part will be evident from the module this function is imported from. Same with all the functions below.